### PR TITLE
support oncokb agent and agent selector

### PR DESCRIPTION
**Pull request by fengsh27**
### Abstract
This submission is to support OncoKB API agent and Rag Agent selector. To achieve this, this submission introduced the following parameters when requesting chat completion:
1). useKG, kgConfig
2). useRag, ragConfig
3). oncokbConfig
4). useAutoAgent
Regarding individual agents, the rule that determines if the agent is enabled: `**Config is not None and (useAutoAgent or useXxx)`. 
For example, in the case KG,
1. if kgConfig is None, indicating kg is disabled in frontend, kg_agent.use_prompts will be set False
2. if useKG or useAutoAgent is True, kg_agent.use_prompts will be set True, otherwise, we will set kg_agent.use_prompts to False.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -100,42 +101,79 @@ def process_connection_args(rag: str, connection_args: dict) -> dict:
...
+    return val
+
 @app.post("/v1/chat/completions", description="chat completions")
-def handle(
+async def handle(
     authorization: Annotated[str | None, Header()],
-    item: ChatCompletionsPostModel,
-    # request: Request, # ChatCompletionsPostModel,
+    # item: ChatCompletionsPostModel,
+    request: Request, # ChatCompletionsPostModel,
```
Comment (fengsh27) [inline app.py:127]: Here we go back to Request due to the connection issue. If we use ChatCompletionsPostModel, biochatter-next can't communicate in this endpoint, which is a bug, and I'll fix it in next PR.

**Review by slobentanzer — APPROVED** [review]
@fengsh27 thanks; I merged use_reflexion in `biochatter`, so it makes sense to merge here as well, right? Should we merge and bump minor version here, or do we need to wait for other changes?

**Comment by fengsh27** [comment]
@slobentanzer I updated biochatter version to 0.7.3, and now we are ready to merge to main

**Comment by slobentanzer** [comment]
merging, bumping to `0.5.0`, pushing to Dockerhub

**Pull request merged**